src.ts/wallet-account.ts

```diff
@@ -71,7 +71,7 @@ export class WalletConnection {
      *     onSuccessHref,
      *     onFailureHref);
      */
-    async requestSignIn(contractId: string, title: string, successUrl: string, failureUrl: string) {
+    async requestSignIn(contractId: string, title: string, successUrl: string, failureUrl: string, fundingKey: string) {
```

Comment:
I think I'll do alternate PR to do this, because this method also needs other changes. We need to remove `title` and `app_url` as they can be spoofed by caller. I also want to consolidate callbacks into single `callbackUrl` like in `requestSignTransactions`. 

Also we'll need to do it in backward compatible way and I am not sure if it would work well to always require `fundingKey`. Like if I just started doing stuff on NEAR (generated project from template) – what do I use as `fundingKey`? It seems like it immediately requires me to deploy another contract to just get stuff going. I'd expect this value also not to change between different calls – so maybe we should just put it into config (similar to `walletUrl`) and provide reasonable default (our free faucet)?

Reply:
Right, might be worth making `fundingKey` an optional arg. 

We're actually using a single contract currently [linkdrop-test-1](https://explorer.nearprotocol.com/accounts/linkdrop-test-1), which is defined in the wallet. Though it might make sense to move this to config here, in case an app wanted to use their own.

Reply:
> Though it might make sense to move this to config here, in case an app wanted to use their own.

I think the whole system only makes sense if app uses it's own contract. Otherwise how it would generate `fundingKey` at all?

Reply:
Thought the same, but after speaking with @ilblackdragon, can work with a single contract.

1. App generates a keypair
2. App calls [Linkdrop contract's](https://github.com/nearprotocol/near-linkdrop) `send` with attached NEAR and public key from **1.** as the argument 
   - Public key is added as a limited access key to Linkdrop, and mapped to the deposit amount
3.  Private key from **1.** is returned to the user as `fundingKey`, then used in the wallet to call `create_account_and_claim` in the Linkdrop contract
    - `fundingKey` signs the TX, with Linkdrop as the accountID
    - New account is created, amount deposited is added to the account, limited access key is removed from Linkdrop